### ISSUE TITLE
Optimize JPanel's

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#277]
Following #276 idea, I like to do the same for JPanel created by some inspections. Currently each option is manually declared on the own inspection class. I don't know if we can change that, but we can make the readability better.

And it is my test result:

```java
public JComponent createOptionsPanel() {
    return OptionsComponent.create((component) -> {
        component.createCheckbox("Report empty() usage", 
            this.optionEmptyUsage, 
            value -> this.optionEmptyUsage = value);

        component.createCheckbox("Suggest to use count()-comparison", 
            this.optionSuggestUseCount, 
            value -> this.optionSuggestUseCount = value);

        component.createCheckbox("Suggest to use null-comparison", 
            this.optionSuggestToUseNullComparison, 
            value -> this.optionSuggestToUseNullComparison = value);
    });
}
```

As I am not a Java programmer, then I have some limitations. For instance, the lambda parameter `component` receives an instance of  `OptionComponent`, that will be used to create each option on inspection. But it receives three parameters: the option title, the default value and a lambda that will be called when the option is changed on IDE. I don't know if I can optimize that to receive only a reference of default value (seems that Java don't have something like pass-as-reference).

**Comment by funivan** [comment]
Sometimes panels are complicated. And it is better to create external class.

**Comment by rentalhost** [comment]
I am working on that just now. I have finished, now I am having some issues as, for instance, the generated checkbox works but not affects or save preferences. haha I am investigating it then I will make a PR about ~1h (else, I will continues tomorrow).

**Comment by rentalhost** [comment]
Argh... I can't do it work. As Java have no support to reference, then I need create an Atomic type (or Mutable), but too for some reason, it change the instance value correctly but IDE can't see that. Each time that I open the settings it creates a new instance (even when it is static!). I'll push to someone check the current code, but for now I will not make a PR.

**Comment by rentalhost** [comment]
About the issue that I have mentined before, I found the cause. Seems that the Intellij SDK can handle only configurations of type `boolean` and `List<>`, and I am trying to use `AtomicBoolean` and similar (if it was possible, this would greatly reduce the code as well).

**From (what I have did after found this problem):**

```java
// boolean optionAnalyzeInterfaces
component.createCheckbox("Analyze interfaces", 
    optionAnalyzeInterfaces, 
    (isSelected) -> optionAnalyzeInterfaces = isSelected);
```

**To (what I liked to do, but don't worked):**

```java
// AtomicBoolean optionAnalyzeInterfaces
component.createCheckbox("Analyze interfaces", optionAnalyzeInterfaces);
```
